daemon/images/image_exporter.go

```diff
@@ -4,17 +4,24 @@ import (
 	"context"
 	"io"
 
+	"github.com/docker/docker/errdefs"
+	"github.com/pkg/errors"
+
 	"github.com/containerd/log"
 	"github.com/docker/docker/container"
 	"github.com/docker/docker/image/tarexport"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 // ExportImage exports a list of images to the given output stream. The
 // exported images are archived into a tar when written to the output
 // stream. All images with the given tag and all versions containing
 // the same tag are exported. names is the set of tags to export, and
 // outStream is the writer which the images are written to.
-func (i *ImageService) ExportImage(ctx context.Context, names []string, outStream io.Writer) error {
+func (i *ImageService) ExportImage(ctx context.Context, names []string, platform *ocispec.Platform, outStream io.Writer) error {
+	if platform != nil {
+		return errdefs.NotImplemented(errors.New("platform parameter is not supported with the graphdriver image store"))
```

Comment:
This probably should be a `ErrInvalidParameter` here, otherwise the API would return a `501`.


Well, either that, or (maybe more "correct"?) we must somehow catch uses of `platform` in the API if the containerd image-store is not enabled, and return an "invalid parameter" there 🤔

Reply:
Not needed after implementing it! 😅